astropy/coordinates/builtin_frames/uvw.py

```diff
@@ -0,0 +1,156 @@
+    It enables easily calculating elevation of a field with e.g.
+
+    #>>> coord = astropy.coordinates.SkyCoord(ra=..., dec=..., frame=UVW(location=..., obstime=..., phase=...))
+    #>>> elevation = coord.transform_to(astropy.coordinates.AltAz).alt
+
+    """
+
+    frame_specific_representation_info = {
+        'cartesian': [RepresentationMapping('x', 'u'),
+                      RepresentationMapping('y', 'v'),
+                     RepresentationMapping('z','w')],
+    }
+    default_representation = CartesianRepresentation
+
+    obstime = TimeAttribute(default=None)
+    location = EarthLocationAttribute(default=None)
+    phase = CoordinateAttribute(ICRS,default=None)
+
+    def __init__(self, *args, **kwargs):
+        super(UVW, self).__init__(*args, **kwargs)
+@frame_transform_graph.transform(FunctionTransform, ITRS, UVW)
+def itrs_to_uvw(itrs_coo, uvw_frame):
+    '''Defines the transformation between ITRS and the UVW frame.'''
+    is_unitspherical = (isinstance(itrs_coo.data, UnitSphericalRepresentation) or
+                        itrs_coo.cartesian.x.unit == u.one)
+    lon, lat, height = uvw_frame.location.to_geodetic('WGS84')
+    #local sidereal time
+    lst = AltAz(alt=90*u.deg,az=0*u.deg,location=uvw_frame.location,obstime=uvw_frame.obstime).transform_to(ICRS).ra
+    #hour angle
+    ha = (lst - uvw_frame.phase.ra).to(u.radian).value
+    dec = uvw_frame.phase.dec.to(u.radian).value
+    lonrad = lon.to(u.radian).value - ha
+    latrad = dec
+    sinlat = np.sin(latrad)
+    coslat = np.cos(latrad)
+    sinlon = np.sin(lonrad)
+    coslon = np.cos(lonrad)
+    north = [-sinlat*coslon,
+                      -sinlat*sinlon,
+                      coslat]
+    east = [-sinlon,coslon,0]
+    up = [coslat*coslon,coslat*sinlon,sinlat]
+    R = np.array([east,north,up])
+    if is_unitspherical:
+        p = itrs_coo.cartesian.xyz.value
+        diff = p
+        penu = R.dot(diff)
+        rep = CartesianRepresentation(x = u.Quantity(penu[0],u.one,copy=False),
+                                     y = u.Quantity(penu[1],u.one,copy=False),
+                                     z = u.Quantity(penu[2],u.one,copy=False),
+                                     copy=False)
+    else:
+        p = itrs_coo.cartesian.xyz
+        p0 = ITRS(*uvw_frame.location.geocentric,obstime=uvw_frame.obstime).cartesian.xyz
+        diff = (p.T-p0).T
+        penu = R.dot(diff)
+        rep = CartesianRepresentation(x = penu[0],
+                y = penu[1],
+                z = penu[2],
+                copy=False)
+
+    return uvw_frame.realize_frame(rep)
+
+@frame_transform_graph.transform(FunctionTransform, UVW, ITRS)
+def uvw_to_itrs(uvw_coo, itrs_frame):
+    lon, lat, height = uvw_coo.location.to_geodetic('WGS84')
+    lst = AltAz(alt=90*u.deg,az=0*u.deg,location=uvw_coo.location,obstime=uvw_coo.obstime).transform_to(ICRS).ra
```

Comment:
Can't this just be computed more easily using `Time.sidereal_time`? http://docs.astropy.org/en/stable/api/astropy.time.Time.html#astropy.time.Time.sidereal_time

Reply:
Didn't know of that method. Will definitely use it.

Reply:
There is a notable problem which I can't understand.
Pytest is failing at `lst = time.sidereal_time('mean',10*u.deg)` with the below error, but when I run these two lines in the shell it works fine...
Pytest doesn't seem to like things.
```
time = at.Time("2017-01-26T17:07:00.000",format='isot',scale='utc')
lst = time.sidereal_time('mean',10*u.deg)
```
produces `IERSRangeError: (some) times are outside of range covered by IERS table.`
It happens because in function `_check_interpolate_indices(self, indices_orig, indices_clipped, max_input_mjd)`